zerver/migrations/0060_move_avatars_to_be_uid_based.py

```diff
@@ -60,15 +60,28 @@ def move_avatars_to_be_uid_based(apps, schema_editor):
         bucket_name = settings.S3_AVATAR_BUCKET
         bucket = conn.get_bucket(bucket_name, validate=False)
         for user_profile in user_profile_model.objects.filter(avatar_source=u"U"):
-            bucket.copy_key(user_avatar_path(user_profile) + ".original",
-                            bucket,
-                            user_avatar_hash(user_profile.email) + ".original")
-            bucket.copy_key(user_avatar_path(user_profile) + "-medium.png",
-                            bucket,
-                            user_avatar_hash(user_profile.email) + "-medium.png")
-            bucket.copy_key(user_avatar_path(user_profile),
-                            bucket,
-                            user_avatar_hash(user_profile.email))
+            uid_hash_path = user_avatar_path(user_profile)
+            email_hash_path = user_avatar_hash(user_profile.email)
+            if bucket.get_key(uid_hash_path):
+                continue
+            if not bucket.get_key(email_hash_path):
```

Comment:
I am not sure about having only `email_hash_path` checked here, since i do not know whether the avatars were always uploaded having three variants namely
* `.original`
* `-medium.png`
* just the name `email_hash_path`
What i mean is can there be a case where anyone of these is missing because i do not have exception handling in place for that since that's a situation that could not arise from the code that i currently see.